crates/parser/src/ast.rs

```diff
@@ -0,0 +1,174 @@
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Copy, Clone)]
+pub enum Operator {
+    Equal,
+    NotEqual,
+    IsAbove,
+    IsAtLeast,
+    IsBelow,
+    IsAtMost,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct Comparison {
+    pub op: Operator,
+    pub target_value: u64,
+}
+
+pub type ParaId = u16;
+pub type NodeName = String;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(tag = "fn", content = "args")]
+pub enum AssertionKind {
+    IsUp {
+        node_name: NodeName,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    ParaIsRegistered {
+        node_name: NodeName,
+        para_id: ParaId,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    ParaBlockHeight {
+        node_name: NodeName,
+        para_id: ParaId,
+        op: Operator,
+        target_value: u64,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    ParaRuntimeUpgrade {
+        node_name: NodeName,
+        para_id: ParaId,
+        file_or_uri: String,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    ParaRuntimeDummyUpgrade {
+        node_name: NodeName,
+        para_id: ParaId,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    Histogram {
+        node_name: NodeName,
+        metric_name: String,
+        op: Operator,
+        target_value: u64,
+        buckets: String,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    Report {
+        node_name: NodeName,
+        metric_name: String,
+        op: Operator,
+        target_value: u64,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    LogMatch {
+        node_name: NodeName,
+        match_type: String,
+        pattern: String,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    Trace {
+        node_name: NodeName,
+        span_id: String,
+        pattern: String,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    SystemEvent {
+        node_name: NodeName,
+        match_type: String,
+        pattern: String,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    CustomJs {
+        node_name: NodeName,
+        file_path: String,
+        custom_args: Option<String>,
+        #[serde(flatten)]
+        cmp: Option<Comparison>,
+        #[serde(with = "optional_timeout")]
+        timeout: Option<Duration>,
+    },
+    CustomSh {
+        node_name: NodeName,
+        file_path: String,
```

Comment:
```suggestion
        file_path: PathBuf,
```

Reply:
Thanks, I will change here and in CustomJs